login-utils/last.c

```diff
@@ -394,8 +394,7 @@ static int list(const struct last_control *ctl, struct utmpx *p, time_t logout_t
 	/*
 	 *	uucp and ftp have special-type entries
 	 */
-	utline[0] = 0;
-	strncat(utline, p->ut_line, sizeof(utline) - 1);
+	xstrncpy(utline, p->ut_line, sizeof(utline));
```

Comment:
The ideal solution will be to use mem2strcpy(), because p->ut_line does not have to be terminated by zero.

Reply:
Changed to mem2strcpy() in commit 28c5ea527c97bc0cf7005545b44bbdcbad442ddd.

And all tests seem to pass when running locally.